### ISSUE TITLE
Pasting into TextField in Text Fields demo in Gallery doesn't work on iOS

### ISSUE THREAD

**Issue by alardizabal** [flutter/flutter#11671]
On iOS, pasting into a text field doesn't work. The validator returns that the field is empty.

![img_0464](https://user-images.githubusercontent.com/3614291/29438842-627eb00c-8387-11e7-9050-761aa48d6079.PNG)

```
[✓] Flutter (on Mac OS X 10.12.6 16G29, locale en-US, channel master)
    • Flutter at /Users/albertlardizabal/dev/flutter
    • Framework revision 60d9479110 (11 minutes ago), 2017-08-17 19:23:59 -0400
    • Engine revision a246501917
    • Tools Dart version 1.25.0-dev.11.0

[✓] Android toolchain - develop for Android devices (Android SDK 26.0.0)
    • Android SDK at /Users/albertlardizabal/Library/Android/sdk
    • Platform android-26, build-tools 26.0.0
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_112-release-b06)

[✓] iOS toolchain - develop for iOS devices (Xcode 8.3.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 8.3.3, Build version 8E3004b
    • ios-deploy 1.9.0
    • CocoaPods version 1.2.1

[✓] Android Studio
    • Android Studio at /Applications/Android Studio 3.0 Preview.app/Contents
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-915-b01)

[✓] Android Studio (version 2.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Java version OpenJDK Runtime Environment (build 1.8.0_112-release-b06)

[✓] IntelliJ IDEA Community Edition (version 2017.2.2)
    • Flutter plugin version 16.0
    • Dart plugin version 172.3757.32

[✓] Connected devices
    • Posse iPhone 6s • ac24c3de3cbe39efed542095769ce83fc89630f7 • ios • iOS 10.3.3
```

**Comment by eseidelGoogle** [comment]
This shows up in the Hamilton App as well.  @cbracken can you take a look?

**Comment by cbracken** [comment]
Will do!

**Comment by cbracken** [comment]
Quick update: looks like this isn't password-specific. I get the same behaviour when pasting into a field with `obscureText: false`.

**Comment by cbracken** [comment]
Looks like `TextSelectionControls.handlePaste` never triggers an update to the underlying text field value within the framework. 

As is, `handlePaste` calls updates its delegate overlay by setting `TextSelectionOverlay.textEditingValue` to the new value, which then pushes an update to the engine. On Android, a change to the text triggers a `restartInput` call on the input manager, which results in the engine is firing an `TextInputClient.updateEditingState` message back into the framework, which then triggers the necessary updates to the text field contents; on iOS, the engine isn't firing this update back (on the assumption that if the framework requested an update, it already knows about it).

**Comment by cbracken** [comment]
Got a two-line patch to fix this. Banging up a test first.

**Comment by cbracken** [comment]
Fix rolled into framework in #11707
